accel/tcg/log_instr.c

```diff
@@ -964,8 +1037,46 @@ void qemu_log_instr_extra(CPUArchState *env, const char *msg, ...)
     va_end(va);
 }
 
+void qemu_log_instr_flush(CPUArchState *env)
+{
+    cpu_log_instr_state_t *cpulog = get_cpu_log_state(env);
+    size_t curr = cpulog->ring_head;
+    cpu_log_instr_info_t *iinfo;
+
+    if ((cpulog->flags & QEMU_LOG_INSTR_FLAG_BUFFERED) == 0)
+        return;
+
+    do {
+        curr = (curr + 1) % cpulog->instr_info->len;
+        iinfo = &g_array_index(cpulog->instr_info, cpu_log_instr_info_t, curr);
+        if (iinfo->flags & LI_FLAG_VALID) {
+            trace_format->emit_entry(env, iinfo);
```

Comment:
Does this also need to call emit_start() and emit_end()?

Reply:
Definitely need to call `emit_entry_event()` instead, I missed it